app/code/Magento/Backend/Controller/Adminhtml/Index/GlobalSearch.php

```
@@ -61,34 +87,73 @@ public function execute()
     }
+
+    /**
+     * retrieve links to certain entities in the global search
+     *
+     * @return array
+     */
+    private function getPreviewItems()
+    {
+        $result = [];
+        $query = $this->getRequest()->getParam('query', '');
+        foreach ($this->previewModules as $previewConfig) {
+            if ($previewConfig['acl'] && !$this->_authorization->isAllowed($previewConfig['acl'])) {
+                continue;
+            }
+            if (!$previewConfig['url'] || !$previewConfig['text']) {
+                continue;
+            }
+            $result[] = [
+                'url' => $this->getUrl($previewConfig['url']).'?search='.$query,
+                'name' => __($previewConfig['text'], $query)
+            ];
+        }
+        return $result;
+    }
+
+    /**
+     * retrieve all entity items that should appear in global search
+     *
+     * @return array
+     */
+    private function getSearchItems()
+    {
+        $items = [];
+        $start = $this->getRequest()->getParam('start', 1);
+        $limit = $this->getRequest()->getParam('limit', 10);
+        $query = $this->getRequest()->getParam('query', '');
+        foreach ($this->_searchModules as $searchConfig) {
+            if ($searchConfig['acl'] && !$this->_authorization->isAllowed($searchConfig['acl'])) {
+                continue;
+            }
+
+            $className = $searchConfig['class'];
+            if (empty($className)) {
+                continue;
+            }
+            try {
+                $searchInstance = $this->itemsFactory->create($className);
+                $results = $searchInstance->setStart(
+                    $start
+                )->setLimit(
+                    $limit
+                )->setQuery(
+                    $query
+                )->getResults();
+                $items = array_merge_recursive($items, $results);
+
+            } catch (\LogicException $exception) {
+                continue;
```

exception swallowing is a bad practice. 
at least we need to log critical message here. 
But in general, LogicalException should not be just swallowed and silently continue script execution 

if you don't know in this specific case how to handle Exception, maybe make sense not to catch it at all